deps/src/singular.cpp

```diff
@@ -53,6 +75,46 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     singular_define_ideals(Singular);
     singular_define_matrices(Singular);
 
+    Singular.method("call_interpreter", [](std::string input) {
+        // set callbacks
+        auto default_print = PrintS_callback;
+        auto default_error = WerrorS_callback;
+        auto default_warning = WarnS_callback;
+
+        PrintS_callback = PrintS_for_julia;
+        WerrorS_callback = WerrorS_for_julia;
+        WarnS_callback = WarningS_for_julia;
+
+        // cleanup return strings
+        singular_return.erase();
+        singular_error.erase();
+        singular_warning.erase();
+
+        // call interpreter
+        std::string  input_str = input + "\nreturn();";
+        const char * input_string = input_str.c_str();
+        char *       ost = (char *)omalloc0(strlen(input_string) + 1);
+        strcpy(ost, input_string);
```

Comment:
```suggestion
        memcpy(ost, input_str.c_str(), input_str.length() + 1);
```

Reply:
Actually, we could do `strcpy(ost, input_str.c_str());`, too. It might be slightly slower.